### PR TITLE
feat(webapp): Scoring for fastest answer and display of time taken

### LINKED ISSUE #12: As a Quizmaster - I want to see the time taken for each response next to the answer
**As a Quizmaster**
I want to see the time taken for each response next to the answer so that I can see how fast people are

### LINKED ISSUE #13: As a Participant - I want to see the time taken for me to submit the answer to the current question
**As a Participant**
I want to see the time taken for me to submit the answer to the current question so I can see how fast I was

### PR DESCRIPTION
**Pull request by richielewis**
Added the following changes
- Increased answer time to 120 seconds, currently hardcoded but eventually will be an option
- Remaining response times displayed on screen of participant
- Remaining response times displayed on quiz master screen
- Remaining response times used to allocate bonus point to fastest answer

resolves #12 
resolves #13

### CODE REVIEW

File: quizmaster-client-app/src/components/QuizHosting/QuestionResponse.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ interface QuestionResponse {
   answer: string;
   name: string;
   id: string;
+  answerTime: number;
```
Comment (calumtomeny) [inline quizmaster-client-app/src/components/QuizHosting/QuestionResponse.ts:5]: Maybe change to include unit i.e. `answerTimeInSeconds`

**Review by calumtomeny — COMMENTED** [review]
Looks good, just a few comments.

File: quizmaster-client-app/src/components/QuizParticipating/QuestionResponder.tsx [hunk 2]
```diff
@@ -76,18 +80,26 @@ export default function QuestionResponder() {
...
+ 
     function progress() {
-      setTimeLeftAsAPercentage((oldCompleted) => {
-        return Math.max(oldCompleted - 10, 0);
-      });
+      if(!answerSubmitted){
+        setTimeLeftAsAPercentage((oldCompleted) => {
+          let increment = 100*(Date.now() - startTime)/(totalTimeInSeconds*1000)
+          return Math.max(100 - increment, 0);
+          //return Math.max(oldCompleted - 1, 0);
```
[the remark this answers is not in the document]
Comment (richielewis) [reply quizmaster-client-app/src/components/QuizParticipating/QuestionResponder.tsx]: Done

File: quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx [hunk 3]
```diff
@@ -109,31 +117,72 @@ export default function QuizHoster() {
...
+ 
     function progress() {
-      setTimeLeftAsAPercentage((oldCompleted) => {
-        return Math.max(oldCompleted - 10, 0);
-      });
+      if(!answersSubmitted){
+        setTimeLeftAsAPercentage((oldCompleted) => {
+          let increment = 100*(Date.now() - questionStartTime)/(totalTimeInSeconds*1000)
+          return Math.max(100 - increment, 0);
+          //return Math.max(oldCompleted - 1, 0);
```
[the remark this answers is not in the document]
Comment (richielewis) [reply quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx]: Deleted

File: quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx [hunk 4]
```diff
@@ -109,31 +117,72 @@ export default function QuizHoster() {
     }
   };
 
+  function getContestantScoreForRound(scores: ContestantScore[],contestantId: String): number {
+    let score = 0;
+    scores.forEach(contestant => {
```
[the remark this answers is not in the document]
Comment (richielewis) [reply quizmaster-client-app/src/components/QuizHosting/QuizHoster.tsx]: Done

File: quizmaster-client-app/src/components/QuizHosting/QuestionResponse.ts (hunk 1 above)
Comment (richielewis) [reply quizmaster-client-app/src/components/QuizHosting/QuestionResponse.ts]: changed to answerTimeLeftAsAPercentage as this is what it contains

**Review by calumtomeny — DISMISSED** [review]

**Review by calumtomeny — COMMENTED** [review]
Looks good.

**Review by calumtomeny — APPROVED** [review]
Looks good.

**Pull request merged**